### ISSUE TITLE
Save detailed Nmap version information in output

### ISSUE THREAD

**Issue by djcater** [nmap/nmap#1277]
I think when specifying `-v` or maybe at `-vvv` Nmap should save the equivalent of the `nmap -V` output into the .nmap and .xml files (and display it in the terminal).

I've just been trying to diagnose an Nmap problem related to different versions of libpcap, but I can't properly compare historical scan files or scan files from different systems because they don't have the full `nmap -V` output in them (in particular the library versions are not there). This is despite the scans having been run with `-vvv`.

I think this would be useful for investigating problems, and if someone is specifying `-v` or more, then they are probably not going to mind an extra few lines of version information at the top of their output.

What do you think?

**Comment by fyodor** [comment]
Yeah, I agree this would be smart information to add.  It should probably happen in debug mode, and also perhaps when the verbosity (-v) level is at least 3.
